qutip/core/qobj.py

```diff
@@ -296,58 +271,110 @@ class Qobj:
     __array_ufunc__ = None
 
     def _initialize_data(self, arg, dims, copy):
+        self._dims = None
+        self._data = None
```

Comment:
I'd much prefer each branch of the `if` to explicitly set `_data` (and `_dims` if it stays here) rather than relying on falling through to `None`.